enterprise/backend/src/metabase_enterprise/sso/integrations/jwt.clj

```diff
@@ -93,22 +94,34 @@
           user         (fetch-or-create-user! first-name last-name email login-attrs)
           session      (api.session/create-session! :sso user (req.util/device-info request))]
       (sync-groups! user jwt-data)
-      (mw.session/set-session-cookies request (response/redirect redirect-url) session (t/zoned-date-time (t/zone-id "GMT"))))))
+      {:session session, :redirect-url redirect-url, :jwt-data jwt-data})))
 
 (defn- check-jwt-enabled []
   (api/check (sso-settings/jwt-enabled)
     [400 (tru "JWT SSO has not been enabled and/or configured")]))
 
+(defn ^:private generate-response-token
+  [session jwt-data]
+  (validation/check-embedding-enabled)
+  (response/response {:id (:id session)
+                      :exp (:exp jwt-data)
+                      :iat (:iat jwt-data)}))
+
 (defmethod sso.i/sso-get :jwt
-  [{{:keys [jwt redirect]} :params, :as request}]
+  [{{:keys [jwt redirect token]
+     :or   {token false}} :params
+    :as                                             request}]
   (premium-features/assert-has-feature :sso-jwt (tru "JWT-based authentication"))
   (check-jwt-enabled)
   (if jwt
-    (login-jwt-user jwt request)
-    (let [idp (sso-settings/jwt-identity-provider-uri)
+    (let [{:keys [session redirect-url jwt-data]} (login-jwt-user jwt request)]
+      (if token
+        (generate-response-token session jwt-data)
+        (mw.session/set-session-cookies request (response/redirect redirect-url) session (t/zoned-date-time (t/zone-id "GMT")))))
```

Comment:
This is a lot of `if`s and `let`s; I'd extract the two branches of `if token` into helper fns.